packages/slice-machine/src/hooks/useModelStatus.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  computeModelStatus,
+  FrontEndModel,
+  ModelStatus,
+} from "@lib/models/common/ModelStatus";
+import { getAuthStatus } from "@src/modules/environment";
+import { AuthStatus } from "@src/modules/userContext/types";
+import { SliceMachineStoreType } from "@src/redux/type";
+import { useSelector } from "react-redux";
+import { useNetwork } from "./useNetwork";
+
+export interface ModelStatusInformation {
+  modelsStatuses: ModelsStatuses;
+  authStatus: AuthStatus;
+  isOnline: boolean;
+}
+
+// Slices and Custom Types needs to be separated as Ids are not unique with each others.
+export type ModelsStatuses = {
```

Comment:
why create a new object, when the model could be extended, seems like there will be two objects to mange ?

Reply:
Good catch, this wasn't necessary at all to create this type.